tests/unit/ssg-module/test_build_yaml.py

```diff
@@ -127,3 +130,67 @@ def test_priority_ordering():
     ordered = ssg.build_yaml.reorder_according_to_ordering(
         to_order, ORDER + ["gaha"], regex=".*ha")
     assert ordered[:2] == ["gaha", "alpha"]
+
+
+@pytest.fixture
+def env_yaml():
+    env_yaml = dict()
+    product_yaml_path = os.path.join(DATADIR, "product.yml")
+    product_yaml = open_raw(product_yaml_path)
+    product_yaml["product_dir"] = os.path.dirname(product_yaml_path)
+    env_yaml["product_cpes"] = ProductCPEs(product_yaml)
+    return env_yaml
+
+
+def test_platform_from_text_unknown_platform(env_yaml):
+    with pytest.raises(ssg.build_cpe.CPEDoesNotExist):
+        ssg.build_yaml.Platform.from_text("something_bogus", env_yaml)
+
+
+def test_platform_from_text_simple(env_yaml):
+    platform = ssg.build_yaml.Platform.from_text("machine", env_yaml)
+    assert platform.to_ansible_conditional() == "ansible_virtualization_type not in [\"docker\", \"lxc\", \"openvz\", \"podman\", \"container\"]"
+    assert platform.to_bash_conditional() == "[ ! -f /.dockerenv ] && [ ! -f /run/.containerenv ]"
+    assert platform.to_xml_element() == b'<?xml version=\'1.0\' encoding=\'utf8\'?>\n<ns0:platform xmlns:ns0="http://cpe.mitre.org/language/2.0" id="machine"><ns0:logical-test operator="AND" negate="false"><ns0:fact-ref name="cpe:/a:machine" /></ns0:logical-test></ns0:platform>'
```

Comment:
I would not write those asserts so literally, as the RHS can change for legitimate reasons, and it would require update of the test code for no reasons.
I suggest to simplify those asserts, so unit tests make sure that things look right, but they the check for the exact "right" to integration tests or system tests that are able to pass even when the syntax changes.

```suggestion
    assert "ansible_virtualization_type" in platform.to_ansible_conditional()
    assert "containerenv" in platform.to_bash_conditional()
    assert platform.to_xml_element()'
```

Reply:
Great!